discoursemap/modules/vulnerability_module.py

```
@@ -275,27 +331,114 @@ def _detect_boolean_sqli(self, response, baseline_response, payload):
             
         return False
     
+    def _detect_time_based_sqli(self, url, payload, baseline_time):
+        """Detect time-based SQL injection by measuring response time"""
+        time_based_indicators = ['SLEEP', 'WAITFOR', 'pg_sleep']
+        
+        if not any(indicator in payload.upper() for indicator in time_based_indicators):
+            return False
+            
+        start_time = time.time()
+        response = self.scanner.make_request(url, params={'q': payload})
```

Ruff (0.12.2)

Local variable `response` is assigned to but never used

Remove assignment to unused variable `response`

(F841)